conifer/backends/xilinxhls/firmware/BDT_rolled.h

```diff
@@ -68,7 +71,11 @@ struct Tree {
       // Only non-leaf nodes do comparisons
       // negative values mean is a leaf (sklearn: -2)
       if(feature[i] >= 0){
-        comparison[i] = x[feature[i]] <= threshold[i];
+        if (!strcmp(splitting_convention,"<=")){
+          comparison[i] = x[feature[i]] <= threshold[i];
+        }else{
+          comparison[i] = x[feature[i]] < threshold[i];
+        }
```

Comment:
Can you provide some side-by-side comparisons of the synthesis (latency, resource usage) of the same model on `master` and this PR? A string comparison seems like something that may be tricky for the HLS tool (it needs to be able to reliably resolve that it's compile-time constant). It would seem to safer to me to do something more similar to what you implemented for the C++ backend with a splitting function. In this case it could be statically defined by the Python writer.